app/js/views/reminders.jsx

```diff
@@ -74,29 +74,45 @@ export default class Reminders extends React.Component {
   onReminder(evt) {
     const reminder = evt.result;
 
-    const reminders = this.state.reminders;
-    reminders.push({
-      id: reminders.length,
-      recipients: reminder.users,
-      content: reminder.action,
-      datetime: reminder.time,
-    });
-
-    this.setState({ reminders });
-
+    // @todo Nice to have: optimistic update.
     this.server.reminders
       .set({
         recipients: reminder.users,
         action: reminder.action,
         due: Number(reminder.time),
       })
+      .then((reminder) => {
```

Comment:
Maybe it would make sense to update the UI before/in parallel to pushing to the server so there is some immediate update in case the network struggles?